src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,60 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  addMovie: (movie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState(false);
+  const [loader, setLoader] = useState(false);
+
+  const handleOnSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    setLoader(true);
+
+    getMovie(query)
+      .then(res => {
+        if ('Error' in res) {
+          return Promise.reject();
+        }
+
+        setMovie({
+          title: res.Title,
+          description: res.Plot,
+          imgUrl: res.Poster === 'N/A' ? (
+            'https://via.placeholder.com/360x270.png?text=no%20preview'
+          ) : (
+            res.Poster
+          ),
+          imdbUrl: `https://www.imdb.com/title/${res.imdbID}`,
+          imdbId: res.imdbID,
+        });
+
+        return Promise.resolve();
```

Comment:
```suggestion
        setMovie({
          title: res.Title,
          description: res.Plot,
          imgUrl: res.Poster === 'N/A' ? (
            'https://via.placeholder.com/360x270.png?text=no%20preview'
          ) : (
            res.Poster
          ),
          imdbUrl: `https://www.imdb.com/title/${res.imdbID}`,
          imdbId: res.imdbID,
        });
```
по ідеї, помилка і так упаде в catch, чи там ерор просто строкою в результаті приходить і резолвиться?